src/include/ipc4/module.h

```diff
@@ -407,10 +407,10 @@ struct ipc4_module_load_library {
 	} data;
 } __packed __aligned(4);
 
-#define IPC4_COMP_ID(x, y)	((x) << 16 | (y))
-#define IPC4_MOD_ID(x) ((x) >> 16)
-#define IPC4_INST_ID(x)	((x) & 0xffff)
-#define IPC4_SRC_QUEUE_ID(x)	(((x) >> 16) & 0xffff)
-#define IPC4_SINK_QUEUE_ID(x)	((x) & 0xffff)
+#define IPC4_COMP_ID(x, y)	((y) << 16 | (x))
+#define IPC4_MOD_ID(x) ((x) & 0xffff)
+#define IPC4_INST_ID(x)	((x) >> 16)
+#define IPC4_SRC_QUEUE_ID(x)	IPC4_MOD_ID(x)
+#define IPC4_SINK_QUEUE_ID(x)	IPC4_INST_ID(x)
```

Comment:
any reason why we are not acessing the bit fields directly here ? I know IPC3 had use bit masks and this is probably why we inherited these conveniene macros.

Reply:
No special reason, just because the bit fields access are the same with IPC4_MOD_ID and IPC4_INST_ID, I will update and keep bit filed access as it was.

Reply:
no, I mean - we dont need the macro for IPC4, we can remove it and access the bit filed directly and let the CC take core of the shifting/masking for us.